Add WSL Instructions

mcdonnnj opened the pull request:
# <!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description
This PR adds instructions for the Windows Subsystem for Linux (WSL) to setup `pyenv` for development usage. There is also clarification about prerequisites and what lines to add to your `.bashrc` for Linux installations.
<!--- Describe your changes in detail -->

## 💭 Motivation and Context
While trying to get INL setup with our development practices on WSL there was a lot of troubleshooting. This should make it easier for a new user to get their development environment in line with our practices more easily.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## 🧪 Testing
I started with a fresh installation of the Ubuntu app on a Windows 10 installation and followed the directions as listed. After completing the `pyenv` installation I was able to install Python versions 3.8.1, 3.7.6, 3.6.10, and 2.7.17 (latest releases of the 3.8, 3,7, 3,6, and 2.7 major versions) with no issues. Those versions should cover reasonable development situations, but are not exhaustive.
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## 🚥 Types of Changes

<!--- What types of changes does your code introduce? -->
<!--- Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

<!--- Go over all the following points, and put an `x` in all the
boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask.
We're here to help! -->

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

mcdonnnj committed:
Add directions to install prerequisites.
Clarify Linux .bashrc additions.
Add Windows Subsystem for Linux specifics.

mcdonnnj committed:
Fix double backtick (`) breaking format.

dav3r commented on CONTRIBUTING.md:
```diff
@@ -56,10 +56,26 @@ eval "$(pyenv init -)"
+chosen to install.
+
+Once you have installed `pyenv` you will need to add the following lines to your
+`.bashrc`:
```
We may want to add something along the lines of: 
"(or whatever file is executed when your particular shell is instantiated)"

mcdonnnj replied:
The installer script outputs instructions for what to add at the end if the command isn't seen [here](https://github.com/pyenv/pyenv-installer/blob/master/bin/pyenv-installer#L89). So maybe instead just note to follow the installer's directions regarding what to add?

dav3r replied:
We can do both- since most people will have bash, what you wrote originally is good.  If they don't run bash, we can direct them to follow the instructions output by the installer- maybe something like so:

> Once you have installed `pyenv` you will need to add the following lines to your
> `.bashrc`:
> ...
> If you are running a shell other than `bash`, follow the instructions output by the `pyenv` installer.

mcdonnnj replied:
Sounds good, I added a line to that effect in 08e7502.

dav3r reviewed (approved):
Thanks for writing this up @mcdonnnj - it should be very helpful for the INL team!  Please consider my one small comment. 📝

jsf9k committed:
Remove an errant space before a period ending a sentence

Also clean up line breaking, since these changes will likely be
backported to skeleton-generic.

jsf9k reviewed (approved):
Thanks for taking care of this @mcdonnnj!

I added a commit that removed an errant space and also cleaned up the line breaking.  I did that because I think these changes should be backported to [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic), so I'd like the formatting to be as clean as possible.  I like for the `*.md` documents to be easily readable as plain text without rendering in Markdown.

mcdonnnj committed:
Add line covering if a user is using a shell other than bash.

jsf9k committed:
Clean up line breakage
